src/main/java/com/powsybl/ws/commons/computation/utils/annotations/PostCompletionAdapter.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ws.commons.computation.utils.annotations;
+
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.support.TransactionSynchronization;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Anis Touri <anis.touri at rte-france.com
+ */
+@Component
+public class PostCompletionAdapter implements TransactionSynchronization {
+    private static final ThreadLocal<List<Runnable>> RUNNABLE = new ThreadLocal<>();
+
+    // register a new runnable for post completion execution
+    public void execute(Runnable runnable) {
+        if (TransactionSynchronizationManager.isSynchronizationActive()) {
+            List<Runnable> runnables = RUNNABLE.get();
+            if (runnables == null) {
+                runnables = new ArrayList<>(Collections.singletonList(runnable));
+            } else {
+                runnables.add(runnable);
+            }
+            RUNNABLE.set(runnables);
+            TransactionSynchronizationManager.registerSynchronization(this);
+            return;
+        }
+        // if transaction synchronisation is not active
+        runnable.run();
+    }
+
+    @Override
+    public void afterCompletion(int status) {
+        List<Runnable> runnables = RUNNABLE.get();
```

Comment:
I am not sure because I don't really know the inner workings of those annotation/runnable stuffs but it is initialised like this : 
`private static final ThreadLocal<List<Runnable>> RUNNABLE = new ThreadLocal<>();`

Why would it be assigned to null ?

But I can add a `if (RUNNABLE != null)` if you want.